### DISCUSSION TITLE
Hide some metadata fields from IIIF manifest for PDF

### DISCUSSION THREAD

**Discussion by andybuki** [jbaiter/pdiiif#18] [Q&A] [answered]
Hi Johannes, i have a following question
i want to hide some metadata fields from IIIF manifest in PDF.

In coverpage.hbs I found a block of code that displays all metadata fields.

```
<dl class="metadata">
          {{#each metadata}}
            <div class="propertyvalue">
              <dt class="label">
                {{this.[0]}}
              </dt>
              <dd class="value">
                {{#ifArray this.[1]}}
                  <ul>
                    {{#each this.[1]}}
                      <li>
                        {{#sanitizeHtml this}}{{/sanitizeHtml}}
                      </li>
                    {{/each}}
                  </ul>
                {{else}}
                  {{#sanitizeHtml this.[1]}}{{/sanitizeHtml}}
                {{/ifArray}}
              </dd>
            </div>
          {{/each}}
        </dl>
```

Is it possible to add some if statement to hide concrete name in metadata?

**Comment by jbaiter** [comment]
Sure, this can easily be accomplished by using a handlebars conditional that checks the label (`this.[0]`) and skips rendering if it fulfills your criterion.
See here for a handlebars primer: https://handlebarsjs.com/guide/expressions.html
Note that you'll have to run your own instance of pdiiif to override the template.

**Comment by andybuki** [comment]
Hi, i am not sure, that i marked a correct piece of code, that responcible for displaying metadata:

Can you add some example in documentation:

For example not display metadata if name is "creator"

**Comment by jbaiter** [comment, marked as answer]
So the part in the template that you want to modify is located here: https://github.com/jbaiter/pdiiif/blob/main/pdiiif-api/assets/coverpage.hbs#L245-L266

This is a loop over all metadata entries, which are tuples/arrays with two elements, with the first value (`this.[0]`) being the label of the manifest entry, and the second (`this.[1]`) the value/values.
What you want to do is to skip a loop iteration if the key matches your criterion, i.e. it equals `creator`. You can do this by wrapping the `<div class="propertyvalue">...</div>` in an `{{#if ...}}` block that checks your criterion.

As I was writing this comment, I noticed that the [required helpers for comparing strings](https://www.npmjs.com/package/handlebars-helpers#unlesseq) are currently not available in pdiiif templates. I'll ship a version soon that will include them, then you can change your template in `assets/coverpage.hbs` like this:

```handlebars
<dl class="metadata">
  {{#each metadata}}
    {{#unlessEq this.[0] "creator" }}
      <div class="propertyvalue">
        <dt class="label">
          {{this.[0]}}
        </dt>
        <dd class="value">
          {{#ifArray this.[1]}}
            <ul>
              {{#each this.[1]}}
                <li>
                  {{#sanitizeHtml this}}{{/sanitizeHtml}}
                </li>
              {{/each}}
            </ul>
          {{else}}
            {{#sanitizeHtml this.[1]}}{{/sanitizeHtml}}
          {{/ifArray}}
        </dd>
      </div>
    {{/unlessEq}}
  {{/each}}
</dl>
```
